src/test/java/org/broadinstitute/hellbender/utils/io/IOUtilsUnitTest.java

```diff
@@ -41,6 +44,212 @@ public void tearDown() {
+        final Path tmpDirPath            = tmpDir.toPath();
+        final Path outputDataSourcesPath = tmpDirPath.resolve(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ).getFileName());
+
+        // Copy our data sources to the destination folder:
+        try {
+            Files.copy(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ), outputDataSourcesPath);
+        }
+        catch (final IOException ex) {
+            throw new GATKException("Could not copy files for testing!", ex);
+        }
+
+        // Extract the files:
+        IOUtils.extractTarGz( outputDataSourcesPath );
+
+        // Extract them again and explode:
+        IOUtils.extractTarGz( outputDataSourcesPath );
+    }
+
+    @Test
+    public void testExtractTarGzOverwrite() {
+
+        // Prepare our output directory:
+        final File tmpDir = createTempDir("IOUtilsUnitTest_testExtractTarGz");
+        tmpDir.deleteOnExit();
+        final Path tmpDirPath            = tmpDir.toPath();
+        final Path outputDataSourcesPath = tmpDirPath.resolve(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ).getFileName());
+
+        // Copy our data sources to the destination folder:
+        try {
+            Files.copy(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ), outputDataSourcesPath);
+        }
+        catch (final IOException ex) {
+            throw new GATKException("Could not copy files for testing!", ex);
+        }
+
+        // Extract the files:
+        IOUtils.extractTarGz( outputDataSourcesPath );
+
+        // Get the base expected data sources output path:
+        final Path baseExpectedOutputDataSourcesPath = IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_FOLDER);
+
+        // Get the base file path for the output:
+        final Path baseActualOutputDataSourcesPath = outputDataSourcesPath.getParent().resolve(
+                baseExpectedOutputDataSourcesPath.getFileName()
+        );
+
+        // Now compare the files with the files in the known output folder:
+        assertContentsTheSame(baseActualOutputDataSourcesPath, baseExpectedOutputDataSourcesPath);
+
+        // Extract them again and overwrite them:
+        IOUtils.extractTarGz( outputDataSourcesPath, outputDataSourcesPath.getParent(), true );
+
+        // Now compare the files with the files in the known output folder AGAIN:
+        assertContentsTheSame(baseActualOutputDataSourcesPath, baseExpectedOutputDataSourcesPath);
+    }
+
+    @Test
+    public void testCreateFifoFile() {
+
+        // Create an output location for the test files to go:
+        final File tmpDir = createTempDir("IOUtilsUnitTest_testCreateFifoFile");
+        tmpDir.deleteOnExit();
+        final Path tmpDirPath = tmpDir.toPath();
+
+        // Create a FIFO file:
+        final Path fifoFilePath = tmpDirPath.resolve(IOUtils.getPath("FIFOFILE"));
+        IOUtils.createFifoFile( tmpDirPath.resolve(fifoFilePath) );
```

Comment:
`fifoFilePath` was already resolved against `tmpDirPath` at this point, wasn't it?

Reply:
Yup.  Fixed.